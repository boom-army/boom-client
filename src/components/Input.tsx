import React from "react";
import { styled } from "@mui/material/styles";

interface InputProps {
  lg?: any;
  type?: string;
  text?: string;
  value?: string;
  onChange?: any;
  placeholder?: string;
  fullWidth?: boolean;
  hideLabel?: boolean;
}

const Wrapper = styled("div")<{
  lg?: any;
  fullWidth?: any;
  props?: any;
}>((props) => ({
  width: `${(props.fullWidth ? "100%" : "315px") || (props.lg && "100%")}`,
  background: `${props.theme.tertiaryColor2}`,
  padding: "0.2rem 0.4rem",
  display: "flex",
  flexDirection: "column",
  justifyContent: "space-between",
  borderBottom: `1px solid ${props.theme.accentColor}`,
  marginBottom: "2rem",

  input: {
    width: "100%",
    padding: "0.5rem",
    background: "inherit",
    border: "none",
    fontSize: "1rem",
    fontFamily: props.theme.font,
    color: props.theme.primaryColor,
  },

  label: {
    color: props.theme.secondaryColor,
    marginBottom: "2px",
  },

  "label.fullWidth": {
    width: "100%",
  },

  "label.hideLabel": {
    border: "0",
    clip: "rect(0 0 0 0)",
    height: "1px",
    margin: "-1px",
    overflow: "hidden",
    padding: "0",
    position: "absolute",
    width: "1px",
  },
}));

const Input = ({
  lg = false,
  type = "text",
  text,
  value,
  onChange,
  placeholder,
  fullWidth = false,
  hideLabel = false,
}: InputProps) => {
  return (
    <Wrapper lg={lg === false ? "false" : "true"}>
      <label className={hideLabel ? "hideLabel" : ""}>{text}</label>
      <input
        autoComplete="new-password"
        type={type}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
    </Wrapper>
  );
};

export default Input;
