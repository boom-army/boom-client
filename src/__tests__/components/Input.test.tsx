import React from "react";
import Input from "../../components/Input";
import { act, fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { createSerializer } from '@emotion/jest'

expect.addSnapshotSerializer(createSerializer(
  {
    classNameReplacer(className, index) {
      return `css-${index}`
    },
    // includeStyles: false 
  }
));


describe("<Input/> component :", () => {
  test("display <Input/> component ", async () => {
    let rendered = render(<Input />, { wrapper: MemoryRouter });
    expect(rendered).toMatchSnapshot();
  });
});

test("Assertion testing of <Input/> component", () => {
  render(<Input />, { wrapper: MemoryRouter });
  expect(screen.getByRole("textbox")).toBeInTheDocument();
});
