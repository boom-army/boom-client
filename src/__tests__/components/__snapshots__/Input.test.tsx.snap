// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> component : display <Input/> component  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .css-0 {
  width: 315px;
  background: undefined;
  padding: 0.2rem 0.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border-bottom: 1px solid undefined;
  margin-bottom: 2rem;
}

.css-0 input {
  width: 100%;
  padding: 0.5rem;
  background: inherit;
  border: none;
  font-size: 1rem;
}

.css-0 label {
  margin-bottom: 2px;
}

.css-0 label.fullWidth {
  width: 100%;
}

.css-0 label.hideLabel {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<body>
    <div>
      <div
        class="css-0"
        lg="false"
      >
        <label
          class=""
        />
        <input
          autocomplete="new-password"
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": .css-0 {
  width: 315px;
  background: undefined;
  padding: 0.2rem 0.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border-bottom: 1px solid undefined;
  margin-bottom: 2rem;
}

.css-0 input {
  width: 100%;
  padding: 0.5rem;
  background: inherit;
  border: none;
  font-size: 1rem;
}

.css-0 label {
  margin-bottom: 2px;
}

.css-0 label.fullWidth {
  width: 100%;
}

.css-0 label.hideLabel {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

<div>
    <div
      class="css-0"
      lg="false"
    >
      <label
        class=""
      />
      <input
        autocomplete="new-password"
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
